lib/logstash/outputs/kafka.rb

```diff
@@ -312,8 +314,18 @@ def close
   def write_to_kafka(event, serialized_data)
     if @message_key.nil?
       record = ProducerRecord.new(event.sprintf(@topic_id), serialized_data)
+      if !@message_headers.nil?
+        @message_headers.each do |key, value|
+          record.headers().add(event.sprintf(key), event.sprintf(value).to_java_bytes)
+        end
+      end
     else
       record = ProducerRecord.new(event.sprintf(@topic_id), event.sprintf(@message_key), serialized_data)
+      if !@message_headers.nil?
```

Comment:
- If you make `message_headers` config with a default value, you can avoid always `nil?` checks.
- It looks newly introduced lines are identical, so we can move the logic to out of this scope (just before `prepare(record)`)